src/logger/logger.py

```diff
@@ -0,0 +1,16 @@
+from abc import ABC, abstractmethod
+
+
+class Logger(ABC):
```

Comment:
i think it'd make sense to rename `Logger` to `PatternLogger` (and accordingly the module from `logger` to `pattern_logger` and `ConsoleLogger` to `ConsolePatternLogger`) to avoid confusion with the generic [`logging` library](https://docs.python.org/3/library/logging.html).